generated/nidigital/nidigital/capture_waveform.py

```diff
@@ -0,0 +1,21 @@
+import collections.abc
+
+
+class CaptureWaveform(collections.abc.Sequence):
+    '''Used by fetch_capture_waveform() to return the capture waveform data'''
+    def __init__(self, capture_waveform_data):
+        self._capture_waveform_data = capture_waveform_data
+
+    def __repr__(self):
+        return str(self)
```

Comment:
From https://docs.python.org/3.4/reference/datamodel.html#object.__repr__
> If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment).

Usually this means You'll want the `__module__` and `__qualname__` of the class in there along with the args.

E.g.
```py
def __repr__(self):
   return "{}.{}({})".format(self.__class__.__module__, self.__class__.__qualname__, ...)
```

Reply:
> Usually this means You'll want the __module__ and __qualname__ of the class in there along with the args.

This is good to know. I couldn't find any best practice tips for `__repr__` online. I checked a couple of popular external libraries and didn't find them including `__module__`:
```
>>> import numpy as np
>>> n = np.arange(2)
>>> repr(n)
'array([0, 1])'

>>> import pandas as pd
>>> t = pd.CategoricalDtype(categories=['b', 'a'], ordered=True)
>>> repr(t)
"CategoricalDtype(categories=['b', 'a'], ordered=True)"
```

Reply:
Look at internal libraries, as they're probably the best citizens:

```py
>>>import datetime
>>> datetime.datetime(1, 2, 3)
datetime.datetime(1, 2, 3, 0, 0)
>>> import operator
>>> operator.attrgetter("foo")
operator.attrgetter('foo')
```

Although that's not consistent either:

```py
>>> import collections
>>> collections.OrderedDict()
OrderedDict()
```

🤷‍♂️ Since you're defining this same type in several libraries, it might be good to tell the user _which_ library it came from.